File: java/Decoupling-Temporal-Services-with-Nexus/exercise/src/main/java/shared/nexus/ComplianceNexusService.java
package shared.nexus;

// ═══════════════════════════════════════════════════════════════════
//  TODO 3: Create the Nexus service interface (the shared contract)
// ═══════════════════════════════════════════════════════════════════
//
// This is the shared contract between teams — like an OpenAPI spec, but durable.
// Both teams depend on this interface:
//   - Payments team creates a stub from it (in the workflow)
//   - Compliance team implements a handler for it (in the worker)
//
// ── What to add: ────────────────────────────────────────────────
//
//   1. Add @Service annotation to the interface (from io.nexusrpc)
//   2. Add one method: checkCompliance(ComplianceRequest) → ComplianceResult
//   3. Mark that method with @Operation (from io.nexusrpc)
//
// ── Template: ───────────────────────────────────────────────────
//
//   @Service
//   public interface ComplianceNexusService {
//       @Operation
//       ComplianceResult checkCompliance(ComplianceRequest request);
//   }

import compliance.domain.ComplianceRequest;
import compliance.domain.ComplianceResult;
import io.nexusrpc.Operation;
import io.nexusrpc.Service;

// TODO: Add @Service annotation
public interface ComplianceNexusService {

    // TODO: Add @Operation annotation
    ComplianceResult checkCompliance(ComplianceRequest request);
}

File: kotlin/decouple-monolith/solution/src/main/kotlin/shared/nexus/ComplianceNexusService.kt
package shared.nexus

import compliance.domain.ComplianceRequest
import compliance.domain.ComplianceResult
import io.nexusrpc.Operation
import io.nexusrpc.Service
import shared.domain.ReviewRequest

/**
 * Nexus Service interface. The shared contract between the Payments and Compliance teams.
 *
 * Both teams depend on this interface:
 *   - Payments creates a stub from it, inside the caller Workflow
 *   - Compliance implements a handler for it, on its own Worker
 *
 * The Nexus runtime validates every method at Worker startup, so all of them need
 * @Operation, even ones you are not calling yet.
 */
@Service
interface ComplianceNexusService {

    @Operation
    fun checkCompliance(request: ComplianceRequest): ComplianceResult

    @Operation
    fun submitReview(request: ReviewRequest): ComplianceResult
}
